conbench/app/_plots.py

```diff
@@ -186,72 +198,168 @@ def _source(
             means.append(unit_fmt(points[-1], unit))
     else:
         points = [x[key] for x in data]
+        # Note(JP): If `means` is just string-formatted `points` then this is
+        # kind of acknowledging that `points` is always a collection of mean
+        # values. It seems that this is a string value field that is only
+        # ever used for tooltip generation?
         means = [unit_fmt(float(x[key]), unit) for x in data if x[key]]
 
     if formatted:
+        # Note(JP): why would we want to calculate the raw numeric data from
+        # the tooltip string labels?
         points = [x.split(" ")[0] for x in means]
 
     source_data = dict(x=dates, y=points, commits=commits, means=means)
     return bokeh.models.ColumnDataSource(data=source_data)
 
 
-def time_series_plot(history, benchmark, run, height=250, width=1000):
+def time_series_plot(history, benchmark, run, height=380, width=1100):
+
+    # log.info("Time series plot for:\n%s", json.dumps(history, indent=2))
+
     unit = history[0]["unit"]
-    current = [
-        {
-            "mean": benchmark["stats"]["mean"],
-            "message": run["commit"]["message"],
-            "timestamp": run["commit"]["timestamp"],
-        }
-    ]
     with_dist = [h for h in history if h["distribution_mean"]]
     formatted, axis_unit = _should_format(history, unit)
 
-    source = _source(history, unit, formatted=formatted)
-    source_x = _source(current, unit, formatted=formatted)
+    # Note(JP): `history` is an ordered list of dicts, each dict has a `mean`
+    # key which is extracted here by default.
+    source_mean_over_time = _source(history, unit, formatted=formatted)
```

Comment:
This is pedantic but shouldn't this variable name be switched with the variable name `source_mean`? To me, `source_mean` would imply the mean of each benchmark (i.e.`distribution_mean=False`) and `source_mean_over_time` would imply the rolling mean (i.e. `distribution_mean=True`)

Reply:
Ha : ) Yeah, I can see that. I think what I did is a tiny improvement over what was there and I hope it's also consistent.

In my mental model from yesterday `source_mean_over_time` means  "this is the bokeh source object for a mean-over-time plot (where mean is on the high/y axis and time is on the right/x axis)".

I will leave this for now. Curious: when I say "let's plot mean over time", what would you draw? :)

Personally, I try to always say "rolling window <var>" when there's a rolling window analysis involved.